src/snowflake/snowpark/modin/pandas/series.py

```diff
@@ -2420,7 +2424,9 @@ def dt(self):  # noqa: RT01, D200
         """
         # TODO: SNOW-1063347: Modin upgrade - modin.pandas.Series functions
         current_dtype = self.dtype
-        if not is_datetime64_any_dtype(current_dtype):
+        if not is_datetime64_any_dtype(current_dtype) and not is_timedelta64_dtype(
```

Comment:
We should also raise an error if user tries to access timedelta property on datetime column and vice-a-versa.

Reply:
Should we be doing this type validation here, or just letting the query compiler or SQL engine take care of it?

Reply:
Generally it's fine to let SQL engine take care of error handling. But timedelta type is special case as it only exists on client so it makes sense to raise type errors in client layer. modin frontend end vs query compiler, right now we are doing such checks in both places, so either works.

Reply:
Good catch. Done.